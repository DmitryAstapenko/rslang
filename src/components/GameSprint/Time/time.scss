.time-container {
  border: 5px solid black;
  border-radius: 50%;
}
.time-value {
  margin: 5px;
  display: flex;
  align-items: center;
  justify-content: center;
}
