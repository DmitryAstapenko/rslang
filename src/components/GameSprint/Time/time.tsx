import React, { useState, useEffect } from "react";

import "./time.scss";

interface ITimeProps {
  finishGame: any;
}

const Time: React.FunctionComponent<ITimeProps> = (props: ITimeProps) => {
  const { finishGame } = props;
  const [sec, setSec] = useState(999);

  useEffect(() => {
    console.log("TIME");
    const time = setTimeout(() => {
      if (sec === 0) {
        finishGame();
      } else {
        setSec((s) => s - 1);
      }
    }, 1000);
    return () => {
      clearTimeout(time);
    };
  }, [sec, finishGame]);

  return (
    <div className="time-container">
      <h1 className="time-value">{sec}</h1>
    </div>
  );
};

export default Time;
