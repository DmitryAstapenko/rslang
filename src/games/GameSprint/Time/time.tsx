import React, { useEffect, useState, useMemo } from "react";

import "./time.scss";

interface ITimeProps {
  // eslint-disable-next-line
  finishGame: Function;
}

// eslint-disable-next-line
const Time: React.FunctionComponent<ITimeProps> = (props: ITimeProps) => {
  const { finishGame } = props;
  const numberOfSeconds = useMemo(() => 60, []);
  const [sec, setSec] = useState(numberOfSeconds);
  useEffect(() => {
    const time = setTimeout(() => {
      if (sec === 0) {
        finishGame();
      } else {
        setSec(sec - 1);
      }
    }, 1000);
    return () => {
      clearTimeout(time);
    };
  }, [sec, finishGame, setSec]);

  return (
    <div className="time-container">
      <h1 className="time-value">{sec}</h1>
    </div>
  );
};

export default Time;
