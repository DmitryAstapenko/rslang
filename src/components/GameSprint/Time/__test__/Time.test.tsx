import React from "react";
import Time from "../Time";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

interface ITimeProps {
  finishGame: () => void;
}

const setUp = (props:ITimeProps) => shallow(<Time {...props} />);

describe("should render Score component", () => {
  let component:any;
  beforeEach(() => {
    const finishGame = () => {
      console.log("finish");
    };
    component = setUp({finishGame});
  });
  it("should contain current time", () => {
    const wrapper = component.find(".time-value");
    expect(wrapper.length).toBe(1);
  });
  it("should render Time component", () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
