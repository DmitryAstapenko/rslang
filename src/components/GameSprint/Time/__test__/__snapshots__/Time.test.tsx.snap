// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Score component should render Time component 1`] = `
<div
  className="time-container"
>
  <h1
    className="time-value"
  >
    30
  </h1>
</div>
`;
